### ISSUE TITLE
Difficulty building w/ CMake choosing incompatible fmt library versions

### ISSUE THREAD

**Issue by arbor-arthur** [symforce-org/symforce#324]
**Describe the bug**
On my M1 MacBookPro I have a newer incompatible version of library fmt at version 9.1.0, installed with brew. When building, CMake finds this version and the build fails.

I get errors that look like:
```
      In file included from /Users/acshi/symforce_not_building/symforce/opt/factor.cc:6:
      In file included from /Users/acshi/symforce_not_building/symforce/opt/./factor.h:17:
      In file included from /Users/acshi/symforce_not_building/symforce/opt/./values.h:277:
      In file included from /Users/acshi/symforce_not_building/symforce/opt/./values.tcc:6:
      In file included from /opt/homebrew/include/fmt/format.h:48:
      /opt/homebrew/include/fmt/core.h:1735:47: error: cannot pass object of non-trivial type 'const std::vector<sym::Key>' through variadic method; call will abort at runtime [-Wnon-pod-varargs]
        const auto& arg = arg_mapper<Context>().map(FMT_FORWARD(val));
                                                    ^
      /opt/homebrew/include/fmt/core.h:204:26: note: expanded from macro 'FMT_FORWARD'
      #define FMT_FORWARD(...) static_cast<decltype(__VA_ARGS__)&&>(__VA_ARGS__)
                               ^
      /opt/homebrew/include/fmt/core.h:1777:10: note: in instantiation of function template specialization 'fmt::detail::make_value<fmt::basic_format_context<fmt::appender, char>, const std::vector<sym::Key> &>' requested here
        return make_value<Context>(val);
```

I solved this by simply removing the call to `find_package` in `symforce/opt/CMakeLists.txt` and always running the CMake code to download and install this library locally.

**To Reproduce**
On an M1 MBP, brew install (the latest version of) fmt. Clone the symforce repository, `cd` into it, and run `python3 -m pip install .`.

**Expected behavior**
Should build without modifications.

**Environment (please complete the following information):**
 - OS and version: MacOS Ventura 13.3.1
 - Python version: Python 3.9.13
 - SymForce Version: 0.8.0

**Additional context**
cmake version 3.26.3

**Comment by chao-qu-skydio** [comment]
Thanks for reporting, this is a known issue to us. This issue is also coupled with spdlog as it also uses fmt.

**Comment by chao-qu-skydio** [comment]
I think this should be fixed in the master (by restricting fmt version to 8), again a new release is coming soon.

**Comment by aaron-skydio** [comment]
Yeah this should be fixed by f9c21ade57bb920048867d3441efb257e0c7894c
